extension-style/src/main/java/com/mapbox/maps/extension/style/model/ModelExtensionImpl.kt

```diff
@@ -0,0 +1,59 @@
+package com.mapbox.maps.extension.style.model
+
+import com.mapbox.maps.MapboxExperimental
+import com.mapbox.maps.extension.style.StyleContract
+import com.mapbox.maps.extension.style.StyleInterface
+import com.mapbox.maps.extension.style.utils.check
+
+/**
+ * Concrete implementation of ModelPlugin, the plugin is used to add a 3D model to be used in the style.
+ *
+ * This API can also be used for updating a model. If the model id was already added, it gets replaced by the new model.
+ */
+@MapboxExperimental
+class ModelExtensionImpl(private val builder: Builder) : StyleContract.StyleModelExtension {
```

Comment:
could `ModelExtensionImpl` be internal btw? we can remove `@MapboxExperimental` then and leave it for public interface only

Reply:
it can not be internal, otherwise the builder is not accessible. The StyleModelExtension interface doesn't expose these APIs.